modules/Microsoft.Insights/webTests/.test/common/deploy.test.bicep

```
@@ -0,0 +1,67 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'ms.insights.webtests-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'iwtcom'
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableDefaultTelemetry bool = true
+
+// ============ //
+// Dependencies //
+// ============ //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: location
+}
+
+module nestedDependencies 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-nestedDependencies'
+  params: {
+    appInsightName: 'dep-<<namePrefix>>-appi-${serviceShort}'
+    logAnalyticsWorkspaceName: 'dep-<<namePrefix>>-law-${serviceShort}'
+    location: location
+  }
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+module testDeployment '../../deploy.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-test-${serviceShort}'
+  params: {
+    name: '<<namePrefix>>${serviceShort}001'
+    tags: {
+      'hidden-link:${nestedDependencies.outputs.appInsightResourceId}': 'Resource'
+    }
```

Please align tags with other examples:

```suggestion
    tags: {
      Environment: 'Non-Prod'
      Role: 'DeploymentValidation'
    }
```

---

Note, I am of the same opinion and we could remove the entire dependency as it's not referenced anywhere else

---

@eriqua please resolve / delete this comment as it turns out the tag actually must be implemented like @shaban-khan did. The `hidden-link` is a mandatory tag, as weird as this may sound.

---

We could at most argue to introduce a new parameter of type array that is more 'explicit' and is unioned into the tags. However, this would diverge from the RP implementation